extensions/http/src/main/java/jolie/net/HttpProtocol.java

```diff
@@ -856,6 +851,10 @@ private void send_appendGenericHeaders(
 				compression = false;
 			}
 			if ( compression ) {
+				if ( !(Interpreter.getInstance().tracer() instanceof DummyTracer) ) {
```

Comment:
Never do this check, it breaks abstraction. The API of the tracer is designed to be efficient and not require these checks at the same time. That's why we use lambdas. In this case, for example, you can just do:

```
Interpreter.getInstance().tracer().trace( () -> {
    final String traceMessage = encodedContent.content.toString(charset);
    return new ProtocolTraceAction(ProtocolTraceAction.Type.HTTP, "HTTP COMPRESSING MESSAGE", message.resourcePath(), traceMessage, null );
} );
```